source/agora/consensus/validation/Block.d

```diff
@@ -82,22 +85,31 @@ version (unittest)
 *******************************************************************************/
 
 public string isInvalidReason (const ref Block block, Height prev_height,
-    in Hash prev_hash, scope UTXOFinder findUTXO,
-    size_t active_enrollments, scope PayloadChecker checkPayload) nothrow @safe
+    in Hash prev_hash, scope UTXOFinder findUTXO, scope PayloadChecker checkPayload,
+    size_t active_enrollments,
+    Point delegate (Height, ulong) nothrow @safe getValidatorAtIndex = null,
+    Point delegate (const ref Point) nothrow @safe getCommitmentNonce = null,
+    string file = __FILE__, size_t line = __LINE__) nothrow @safe
 {
     import std.algorithm;
+    import std.string;
+    import std.conv;
 
-    if (block.header.height > prev_height + 1)
-        return "Block: Height is above expected height";
-    if (block.header.height < prev_height + 1)
-        return "Block: Height is under expected height";
+    if (block.header.height == 0)
+        return "Block: Genesis block should be validated using isGenesisBlockInvalidReason";
+
+    if (block.header.height != prev_height + 1)
+        return "Block: Height is not one more than previous block";
 
     if (block.header.prev_block != prev_hash)
         return "Block: Header.prev_block does not match previous block";
 
     if (block.txs.length == 0)
         return "Block: Must contain at least one transaction";
 
+    if (block.header.enrollments.length + active_enrollments < Enrollment.MinValidatorCount)
```

Comment:
It seems like this change was accidental?

Reply:
No, I just moved it up in the list of checks to be more efficient.